loguru_mypy/__init__.py

```diff
@@ -0,0 +1,150 @@
+        if res[1] is None:
+            continue
+        elif not res[1].strip():
+            log_msg_expected_args_count += 1
+        else:
+            log_msg_expected_kwargs.append(res[1].strip())
+
+    if log_msg_expected_args_count > call_args_count:
+        ctx.api.msg.fail(
+            f'Missing {log_msg_expected_args_count - call_args_count} '
+            'positional arguments for log message',
+            context=log_msg_expr,
+            code=ERROR_BAD_ARG,
+        )
+        return ctx.default_return_type
+    elif log_msg_expected_args_count < call_args_count:
+        ctx.api.msg.note(
+            f'Expected {log_msg_expected_args_count} but found {call_args_count} '
+            'positional arguments for log message',
+            context=log_msg_expr,
+            code=ERROR_BAD_ARG,
+        )
+        return ctx.default_return_type
+    else:
+        for call_pos, call_arg in enumerate(call_args):
+            if isinstance(call_arg, LambdaExpr) and call_arg.arguments:
+                ctx.api.msg.fail(
+                    f'Expected 0 arguments for <lambda>: {call_pos} arg ',
+                    context=call_arg,
+                    code=ERROR_BAD_ARG,
+                )
+            elif isinstance(call_arg, RefExpr) and isinstance(
+                    call_arg.node, FuncDef) and call_arg.node.arguments:
+                ctx.api.msg.fail(
+                    f'Expected 0 arguments for {call_arg.node.fullname}: {call_arg}',
+                    context=call_arg,
+                    code=ERROR_BAD_ARG,
+                )
+
+    for log_msg_kwarg in log_msg_expected_kwargs:
+        maybe_kwarg_expr = call_kwargs.pop(log_msg_kwarg, None)
+        if maybe_kwarg_expr is None:
+            ctx.api.msg.fail(
+                f'{log_msg_kwarg} keyword argument is missing',
+                context=log_msg_expr,
+                code=ERROR_BAD_KWARG,
+            )
+            return ctx.default_return_type
+        elif isinstance(maybe_kwarg_expr, LambdaExpr) and maybe_kwarg_expr.arguments:
+            ctx.api.msg.fail(
+                f'Expected 0 arguments for <lambda>: {log_msg_kwarg} kwarg ',
+                context=maybe_kwarg_expr,
+                code=ERROR_BAD_KWARG,
+            )
+        elif isinstance(maybe_kwarg_expr, RefExpr) and isinstance(
+                maybe_kwarg_expr.node, FuncDef) and maybe_kwarg_expr.node.arguments:
+            ctx.api.msg.fail(
+                'Expected 0 arguments for '
+                f'{maybe_kwarg_expr.node.fullname}: {log_msg_kwarg}',
+                context=maybe_kwarg_expr,
+                code=ERROR_BAD_KWARG,
+            )
+
+    for extra_kwarg_name in call_kwargs:
+        ctx.api.msg.fail(
+            f'{extra_kwarg_name} keyword argument not found in log message',
+            context=log_msg_expr,
+            code=ERROR_BAD_KWARG,
+        )
+
+    return ctx.default_return_type
+
+
+class LoguruPlugin(Plugin):
+    builtin_severities = (
```

Comment:
Maybe add `trace` and `success` too (those are valid logging methods)?

Reply:
Done